pandas/core/indexes/base.py

```diff
@@ -277,10 +277,6 @@ def __new__(
     ) -> "Index":
 
         from pandas.core.indexes.range import RangeIndex
```

Comment:
We previously delayed the `RangeIndex` import by doing the following:

https://github.com/pandas-dev/pandas/blob/250574eda5302b6941f8c766f520f67226aadad7/pandas/core/indexes/base.py#L185-L191

Not sure if this is still possible but might be worthwhile to look into.